js/__tests__/stats-test.js

```diff
@@ -0,0 +1,457 @@
+/*
+Author: Tony J Huang
+Group: Team Stimulus
+Created At: 3/15/15
+
+Tests for the Stats module. Uses Jest testing framework. See
+the Jest documentation here:
+
+https://facebook.github.io/jest/docs/api.html#content
+*/
+
+jest.dontMock("../stats.js");
+jest.dontMock("../picture.js");
+jest.dontMock("../validate.js");
+jest.dontMock("../underscore-min.js");
+
+// Mock Data
+var Stats = require("../stats.js");
+var Picture = require("../picture.js");
+var _ = require("../underscore-min.js");
+var emptyPictures = [];
+var mockPictures = [new Picture(7, "foo")];
+var mockPictures2 = [new Picture(5, "foo"), 
+  new Picture(7, "bar"), 
+  new Picture(9, "baz"),
+  new Picture(9, "qux")];
+
+// Tests for our stats functions.
+describe("statistics", function() {
+  it("computes the median rating", function() {
+    expect(Stats.medianRating(mockPictures)).toBe(7);
+  });
+
+  it("also computes the median rating", function() {
+    expect(Stats.medianRating(mockPictures2)).toBe(8);
+  });
+
+  it("throws an error on median"), function() {
+    expect(Stats.medianRating(emptyPictures)).toThrow("not an Array");
```

Comment:
youre right, actually that wasnt even the error it throws, but ive updated it. not entirely sure why this test passed.